lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,207 @@
+    const inputFiles = evt.currentTarget.files;
+
+    // files refers to the previous file selection, which will be null the first time OnChange is called
+    const returnedFileList = inputFiles ? combineFileLists(files ?? [], inputFiles) : null,
+        normalizedFiles = returnedFileList?.length ? returnedFileList : null;
+
+    onChangeProp(normalizedFiles);
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj: File) {
+    const closeBtns = Array.from(
+        selectedFilesContainerRef.current?.querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []
+    );
+
+    if (files) {
+      const filesArray = Array.from(files);
+      const idx = findIndex((f) => equals(f, fileObj), filesArray);
+
+      if (idx === files.length - 1) {
+        // if the selected file is the last in the container, set the focus to the next last dismiss button
+        // if selected file is only file, set focus on the input
+        closeBtns.length === 1 ? inputRef.current?.focus() : closeBtns[idx - 1].focus();
+      }
+      else {
+        closeBtns[idx + 1].focus({preventScroll: true});
+      }
+
+      const updatedSelectedFiles = without([fileObj], filesArray),
+          returnedFileList = combineFileLists(updatedSelectedFiles),
+          normalizedFiles = returnedFileList.length ? returnedFileList : null;
+
+      onChangeProp(normalizedFiles);
+    }
+  }
+
+  useEffect(function() {
+    if (inputRef.current) {
+      if (isFileSelected) {
+        inputRef.current.files = files;
+      }
+      else {
+        // there's no way to clear the list via the files prop, but this does it
+        inputRef.current.value = '';
+      }
+    }
+  }, [files]);
+
+  return (
+    <div ref={ref} className={className} >
+      <div className="nx-multi-file-upload__container">
+        <input ref={inputRef}
+               { ...attrs }
+               disabled={disabled}
+               onChange={onChange}
+               id={inputId}
+               className="nx-file-upload__input"
+               type="file"
+               aria-describedby={totalFilesSelectedId}
+               aria-required={isRequired ?? undefined}
+               aria-invalid={showError || undefined}
+               aria-errormessage={showError ? validationErrorId : undefined}
+               aria-controls={selectedFilesContainerId}
+               multiple/>
+        {/* Keynav and screenreaders can ignore the button itself in favor of the <input> */}
+        <NxButton type="button"
+                  aria-hidden={true}
+                  disabled={disabled}
+                  variant="tertiary"
+                  onClick={openPicker}
+                  tabIndex={-1}
+                  className="nx-file-upload__select-btn">
+          Add Files
+        </NxButton>
+        <span id={totalFilesSelectedId} className="nx-multi-file-upload__file-count">
+          {totalFilesSelected}
```

Comment:
> There's still a little bit more redundancy here than I intended. What I wanted was for this element to go away, and for the `nx-file-upload__no-file-message` to be used as the description element. Have that element always present in the DOM but conditionally hidden.

Thanks for the clarification, I implemented it here: [2624904](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/2624904265aed3b09c2ed4c14fd33432aea88b8b). 

2 questions arose from this: 
1) I have left `nx-file-upload__no-file-message` as an `<li>` and not sure if it's ideal. But if it's outside of `nx-multi-file-upload__container__files`, it would need custom CSS to position correctly. Leave it as an `<li>` ?
2) I left the variable names `noFileMessageClassName` and `noFileMessageId` as is. I thought about changing them to something like `fileCountMessageClassName` etc, but then the variable name wouldn't match the classnames assigned to it. Would you prefer to change the variable names?